modules/core/src/main/java/org/apache/ignite/Ignite.java

```diff
@@ -589,6 +589,9 @@ public IgniteSemaphore semaphore(String name, int cnt, boolean failoverSafe, boo
     public IgniteLock reentrantLock(String name, boolean failoverSafe, boolean fair, boolean create)
         throws IgniteException;
 
+    /** */
+    public IgniteLock reentrantLock(String name, boolean fair, boolean create) throws IgniteException;
```

Comment:
why u haven't implelemnted failoverSafe flag ? I wrote about it.

Reply:
I think this is a task for the other issue in Jira:-)